packages/webapp/src/utils/zkproofs/proofs.ts

```diff
@@ -79,12 +86,16 @@ export async function prove
     // NOTE: proof can be verified with
     //       `verify(circuitName, publicSignals, proof)`
 
-    // Probably because they were high, the snarkjs author decided to return a string here,
-    // formatted as '["proofItem1", "proofItem2", ...]["publicSignal1", "publicSignal2", ...]'.
-    const calldata = await snarkjs.plonk.exportSolidityCallData(proof, publicSignals)
-
-    // Clean this fucking mess up.
-    return { proof: JSON.parse(calldata.split("][", 1)[0] + "]").map(BigInt) }
+    // Parse proof outputs
+    const generated_proof: ProofOutput =  { 
+      proof_a: proof["pi_a"].slice(0,2) as [bigint, bigint],
+      proof_b: [
+        [proof["pi_b"][0][1], proof["pi_b"][0][0]],
+        [proof["pi_b"][1][1], proof["pi_b"][1][0]]
+      ] as [[bigint, bigint], [bigint, bigint]],
```

Comment:
This is the Endian difference that needs to be parsed in order to match the solidity verifier.

Reply:
Let's add a comment to that affect in the code.